### ISSUE TITLE
How long does it take to train from scratch?

### ISSUE THREAD

**Issue by monjurulkarim** [matterport/Mask_RCNN#1995]
Hello, can anybody please tell me how long does it take to train the network with ms coco dataset from scratch? It would be great if anybody can give me a reference that shows the training time required by mask rcnn.

**Comment by jqx37** [comment]
Hi. I have some estimated training time from my experience. 
Case 1: On a Quadro M1000M GPU, for an image size of 640*480, one image per each step, it takes roughly 3-4s per step without image augmentation. With image augmentations (3 out of 5 each step), it takes about 5s per step.
Case 2: On an RTX 2080 Max-Q laptop, with the same image size and settings, it takes about 1s per step with augmentations.

**Comment by JordanMakesMaps** [comment]
@jqx37, how did those results compare to starting with pre-trained weights?

**Comment by jqx37** [comment]
> @jqx37, how did those results compare to starting with pre-trained weights?

@JordanMakesMaps, I used my own dataset. I can say the results are great for me. I did a transfer learning.

**Comment by JordanMakesMaps** [comment]
@jqx37, thanks. I saw on a few other threads people were recommending initializing with COCO and then finetune by training 'head', and then 'all' separately. 

I'll check it out, thanks.

**Comment by jqx37** [comment]
@JordanMakesMaps, you are welcome. Yes, you can do it that way with an RTX 2080 Max-q to fine-tune the network, you need a graphic card with larger VRAM. In my case, an RTX 2080 Max-Q works. FYI, on the Quadro M1000M card, you would not be able to train the entire network, only "Heads" can be fitted into the VRAM. For the 8GB RTX 2080 Max-Q, I would be able to train all of the layers with a batch size of 2. For both cards, I fed with 640 by 480 images.

**Comment by JordanMakesMaps** [comment]
Awesome! Thanks for the spec details.

**Comment by jqx37** [comment]
@JordanMakesMaps, You are very welcome.

**Comment by HabibMrad** [comment]
do i need to use AWS?
